src/frost.rs

```diff
@@ -284,22 +286,37 @@ fn generate_shares<R: RngCore + CryptoRng>(
         commitment.0.push(Commitment(SpendAuth::basepoint() * c));
     }
 
+    // Participants' identifiers are generated as follows:
+    // id = H("FROST_id", g^s, i), where:
+    // g is the basepoint,
+    // s is the secret chosen by the dealer, and
+    // i is the index of the participant.
+    // The hash is finalized in the loop that follows.
+    let mut hasher = HStar::default();
+    hasher
+        .update("FROST_id".as_bytes())
+        .update(jubjub::AffinePoint::from(SpendAuth::basepoint() * secret.0).to_bytes());
+
     // Evaluate the polynomial with `secret` as the constant term
     // and `coeffs` as the other coefficients at the point x=share_index,
     // using Horner's method.
     for index in 1..numshares + 1 {
-        let scalar_index = Scalar::from(index as u64);
+        // The actual identifier consists of the first 64 bits of the resulting hash.
+        let id_bytes = hasher.update(index.to_be_bytes()).finalize().to_bytes();
```

Comment:
Why big endian bytes?

Reply:
I saw it here https://github.com/ZcashFoundation/redjubjub/blob/9c9825087194d564b1e390c1a5de6e831b4e5bbd/src/frost.rs#L495 and here https://github.com/ZcashFoundation/redjubjub/blob/9c9825087194d564b1e390c1a5de6e831b4e5bbd/src/frost.rs#L499

Reply:
Aha, I have a theory on why: in frost-dalek, they use SHA512 as their hash, which uses big-endian in their encodings, and frost-dalek uses big-endian byte version of indices when updating the SHA512 state. 

We use BLAKE2b, which use little-endian encodings, so we should probably match for that reason, but also because being consistently little-endian whenever possible is helpful for avoiding bugs. 
https://crypto.stackexchange.com/questions/2099/why-do-all-hash-functions-use-big-endian-data

Reply:
That's a great explanation, thank you. The only reason I went with big endian was that I saw it at the spots I posted.